src/nunit.analyzers.tests/ClassicModelAssertUsage/AreSameClassicModelAssertUsageCodeFixTests.cs

```diff
@@ -0,0 +1,95 @@
+using NUnit.Analyzers.Constants;
+using NUnit.Framework;
+
+namespace NUnit.Analyzers.Tests.ClassicModelAssertUsage
+{
+    [TestFixture]
+    public sealed class AreSameClassicModelAssertUsageCodeFixTests
+    {
+        private static readonly DiagnosticAnalyzer analyzer = new ClassicModelAssertUsageAnalyzer();
+        private static readonly CodeFixProvider fix = new AreSameClassicModelAssertUsageCodeFix();
+        private static readonly ExpectedDiagnostic expectedDiagnostic =
+            ExpectedDiagnostic.Create(AnalyzerIdentifiers.AreSameUsage);
+
+        [Test]
+        public void VerifyGetFixableDiagnosticIds()
+        {
+            var fix = new AreSameClassicModelAssertUsageCodeFix();
+            var ids = fix.FixableDiagnosticIds.ToImmutableArray();
+
+            Assert.That(ids.Length, Is.EqualTo(1), nameof(ids.Length));
+            Assert.That(ids[0], Is.EqualTo(AnalyzerIdentifiers.AreSameUsage), nameof(AnalyzerIdentifiers.AreSameUsage));
+        }
+
+        [Test]
+        public void VerifyAreSameFix()
+        {
+            var code = TestUtility.WrapMethodInClassNamespaceAndAddUsings($@"
+        public void TestMethod()
+        {{
+            var expected = new object();
+            var actual = new object();
+
+            ↓Assert.AreSame(expected, actual);
+        }}");
+            var fixedCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+        public void TestMethod()
+        {
+            var expected = new object();
+            var actual = new object();
+
+            Assert.That(actual, Is.SameAs(expected));
+        }");
+            AnalyzerAssert.CodeFix(analyzer, fix, expectedDiagnostic, code, fixedCode, fixTitle: CodeFixConstants.TransformToConstraintModelDescription);
+        }
+
+        [Test]
+        public void VerifyAreSameFixWithMessage()
+        {
+            var code = TestUtility.WrapMethodInClassNamespaceAndAddUsings($@"
+        public void TestMethod()
+        {{
+            var expected = new object();
+            var actual = new object();
+
+            ↓Assert.AreSame(expected, actual, ""message"");
+        }}");
+            var fixedCode = TestUtility.WrapMethodInClassNamespaceAndAddUsings(@"
+        public void TestMethod()
+        {
+            var expected = new object();
+            var actual = new object();
+
+            Assert.That(actual, Is.SameAs(expected), ""message"");
+        }");
+            AnalyzerAssert.CodeFix(analyzer, fix, expectedDiagnostic, code, fixedCode, fixTitle: CodeFixConstants.TransformToConstraintModelDescription);
+        }
+
+        [Test]
+        public void VerifyAreSameFixWithMessageAndParams()
+        {
+            var code = TestUtility.WrapMethodInClassNamespaceAndAddUsings($@"
+        public void TestMethod()
+        {{
+            var expected = new object();
+            var actual = new object();
+
+            ↓Assert.AreSame(expected, actual, ""message"", Guid.NewGuid());
```

Comment:
An alternative is to put the diagnostic on the identifier. Not saying it is better, just an alternative. If the assert is multiline there will be a bunch of squiggle when having it on the expression. But we can start like this and see if it feels ergonomic.

Reply:
I had not considered how this looks when having a multiline assert (almost all of my asserts are on a single line 😄). I think I'll keep it as it is - for now - as this align with the other analyzers, but it is worth to consider, if we should change them all.